crates/bevy_pbr/src/light_probe/environment_map.wgsl

```diff
@@ -176,14 +177,13 @@ fn environment_map_light(
     // Calculate the Fresnel term `Fc` for the clearcoat layer.
     // 0.04 is a hardcoded value for F0 from the Filament spec.
     let clearcoat_F0 = vec3<f32>(0.04);
-    let Fc = F_Schlick_vec(clearcoat_F0, 1.0, clearcoat_NdotV) * clearcoat;
+    let Fc = F_Schlick_vec(clearcoat_F0, 1.0, clearcoat_NdotV) * clearcoat_strength;
     let inv_Fc = 1.0 - Fc;
-    let clearcoat_radiances = compute_radiances(
-        clearcoat_perceptual_roughness,
-        clearcoat_N,
-        clearcoat_R,
-        world_position,
-        found_diffuse_indirect);
+
+    // We have to copy `clearcoat` here to work around Naga bug
+    // <https://github.com/gfx-rs/wgpu/issues/5593>.
```

Comment:
from the duped issue this doesn't sound like a bug, just an unsupported extension (`unrestricted_pointer_parameters`).

not clear to me if this is an extension we can request / something that's supported in some contexts though.

Reply:
Could we make this work by passing around an actual struct that contains pointers to a base, a base_layer and a clearcoat_layer?

Reply:
You can't put `ptr<function, ...>` types in structs in WGSL. I just tried.

Reply:
Right, that is explicitly denied in the spec. Maybe use a fixed-size array of layers and pass the full ptr and an index into the layer-based functions..?

I don’t know if this would be worthwhile; I can try it and profile later if you don’t want to (I realise I’ve been pretty demanding already)

Reply:
OK, I did that (layers and indices).